subsys/usb/device/class/audio/usb_audio_internal.h

```diff
@@ -165,14 +165,26 @@
 #define GET_VOLUME_MIC(i, prop) DT_PROP_OR(DT_INST(i, COMPAT_MIC), prop, 0)
 #define GET_VOLUME(dev, i, prop) GET_VOLUME_##dev(i, prop)
 
+#define GET_RATE_HP(i) DT_PROP(DT_INST(i, COMPAT_HP), sample_rate_hz)
+#define GET_RATE_MIC(i) DT_PROP(DT_INST(i, COMPAT_MIC), sample_rate_hz)
+#define GET_RATE_HS_HP(i) DT_PROP(DT_INST(i, COMPAT_HS), hp_sample_rate_hz)
+#define GET_RATE_HS_MIC(i) DT_PROP(DT_INST(i, COMPAT_HS), mic_sample_rate_hz)
+#define GET_RATE(dev, i) GET_RATE_##dev(i)
+
+#define GET_INTERVAL_HP(i) DT_PROP(DT_INST(i, COMPAT_HP), polling_interval)
+#define GET_INTERVAL_MIC(i) DT_PROP(DT_INST(i, COMPAT_MIC), polling_interval)
+#define GET_INTERVAL_HS_HP(i) DT_PROP(DT_INST(i, COMPAT_HS), hp_polling_interval)
+#define GET_INTERVAL_HS_MIC(i) DT_PROP(DT_INST(i, COMPAT_HS), mic_polling_interval)
+#define GET_INTERVAL(dev, i) GET_INTERVAL_##dev(i)
+
 #define SYNC_TYPE_HP(i)     3
 #define SYNC_TYPE_MIC(i)    DT_ENUM_IDX(DT_INST(i, COMPAT_MIC), sync_type)
 #define SYNC_TYPE_HS_HP(i)  3
 #define SYNC_TYPE_HS_MIC(i) DT_ENUM_IDX(DT_INST(i, COMPAT_HS), mic_sync_type)
 #define SYNC_TYPE(dev, i) (SYNC_TYPE_##dev(i) << 2)
 
 #define EP_SIZE(dev, i) \
-	((GET_RES(dev, i)/8) * CH_CNT(dev, i) * 48)
+	((GET_RES(dev, i)/8) * CH_CNT(dev, i) * (GET_RATE(dev, i)/1000))
```

Comment:
This doesn't account for sample rates that are not full multiple of 1000, e.g. 44100. However to support 44100 properly also changes in other parts of the class implementation would be needed.

Considering that I am actively working on UAC2 implementation for the new stack, I am not sure if it is worthwhile to try to fix the old USB Audio implementation.

Reply:
On my side I'm actually using 3.4.0, would it be worth targeting this as a change to 3.4.0 instead?  What is the ETA for UAC2 support?

Reply:
I hope is that I will manage to open the PR with UAC2 + explicit feedback sample in January. The implicit feedback sample will follow later (the remaining complexity is on the sample application side, the class is pretty much ready for implicit feedback). I will CC you when I open the PR.